### ISSUE TITLE
Update DOIs from KNB to redirect to view service

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#1083]
---

Author Name: **Matt Jones** (Matt Jones)
Original Redmine Issue: 6793, https://projects.ecoinformatics.org/ecoinfo/issues/6793
Original Date: 2015-07-02
Original Assignee: ben leinfelder

---

In #6530 and #6440, we added features to update DOI registrations, but we still have many originally assigned DOIs that redirect to the raw EML document rather than our landing page for a data set.  We need to fix all of the /AA/ DOI registrations in the KNB and ensure they point to the right View service page.  For DOIs for metadata, that would be the associated /view url for that DOI.  For data files and resource maps, its to the view for the associated metadata.  E.g.,

* Metadata doi:10.5063/AA/nceas.227.15 should redirect to https://knb.ecoinformatics.org/#view/doi:10.5063/AA/nceas.227.15
* Data doi:10.5063/AA/wtyburczy.30.1 should redirect to the same metadata file https://knb.ecoinformatics.org/#view/doi:10.5063/AA/nceas.227.15

Also, when a user updates metadata for a package (but doesn't change the data), the DOI redirect for the data will need to be updated to point to the new metadata.  Let's verify that this is happening automatically in Metacat.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Lauren Walker** (Lauren Walker)
Original Date: 2015-07-02T19:46:17Z

---

MetacatUI 1.7.0 will redirect to the associated metadata view for any given data pid. So managing the EZID registrations for those DOI's, at least, is unnecessary.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **ben leinfelder** (ben leinfelder)
Original Date: 2016-01-25T15:33:08Z

---

Megan Force at Thompson Reuters is reporting our old DOIs still point to XML representations of the metadata rather than the #view landing page.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **ben leinfelder** (ben leinfelder)
Original Date: 2016-01-25T15:48:03Z

---

I was able to use our Metacat admin tool to update an example DOI - doi:10.5063/AA/dpennington.338.2 now points to the #view page.
Running the update routine for all EML format ids now.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **ben leinfelder** (ben leinfelder)
Original Date: 2016-01-25T18:16:24Z

---

Ran the DOI updater for EML 2.0.0, 2.0.1, 2.1.0, and 2.1.1 formats. All EML DOIs should be resolving to the #view endpoint now.

**Comment by taojing2002** [comment]
I tested the tool on the Metacat admin page and it didn't work. I looked at code and found the tool get s`docids` rather than `pids` from `xml_documents` and `xml_revisions` tables. Then it examines the `docid` to see if it starts with `doi`. If the `docid` starts with the `doi`, the tool will update the datacite information for this `docid`. However, `docids` generally don't start with `doi`, so no `docids` were processed. We should look at the `pids` on the `systemmetadata` table.
